iFrom d23062b20b5d65668423f60767d8f78491931634 Mon Sep 17 00:00:00 2001
From: Jean-Paul Saman <package@m2x.nl>
Date: Mon, 11 Jun 2012 11:26:36 +0200
Subject: [PATCH] Qt4: Solve naming conflict with with mingw32mvsc header
 files.

Mingw32mvsc uses small as a type and this conflicts with using it as a variable or class member.
Therefor rename 'small' class member to small_rb and 'small' boolean to 'b_small'.
---
 modules/gui/qt4/components/simple_preferences.cpp |    8 ++++----
 modules/gui/qt4/dialogs/preferences.cpp           |   14 +++++++-------
 modules/gui/qt4/dialogs/preferences.hpp           |    3 ++-
 3 files changed, 13 insertions(+), 12 deletions(-)

diff --git a/modules/gui/qt4/components/simple_preferences.cpp b/modules/gui/qt4/components/simple_preferences.cpp
index 9d67382..82edb7a 100644
--- a/modules/gui/qt4/components/simple_preferences.cpp
+++ b/modules/gui/qt4/components/simple_preferences.cpp
@@ -57,7 +57,7 @@
 /*********************************************************************
  * The List of categories
  *********************************************************************/
-SPrefsCatList::SPrefsCatList( intf_thread_t *_p_intf, QWidget *_parent, bool small ) :
+SPrefsCatList::SPrefsCatList( intf_thread_t *_p_intf, QWidget *_parent, bool b_small ) :
                                   QWidget( _parent ), p_intf( _p_intf )
 {
     QVBoxLayout *layout = new QVBoxLayout();
@@ -67,7 +67,7 @@ SPrefsCatList::SPrefsCatList( intf_thread_t *_p_intf, QWidget *_parent, bool sma
     CONNECT( buttonGroup, buttonClicked ( int ),
             this, switchPanel( int ) );
 
-    short icon_height = small ? ICON_HEIGHT /2 : ICON_HEIGHT;
+    short icon_height = b_small ? ICON_HEIGHT /2 : ICON_HEIGHT;
 
 #define ADD_CATEGORY( button, label, icon, numb )                           \
     QToolButton * button = new QToolButton( this );                         \
@@ -116,7 +116,7 @@ void SPrefsCatList::switchPanel( int i )
  * The Panels
  *********************************************************************/
 SPrefsPanel::SPrefsPanel( intf_thread_t *_p_intf, QWidget *_parent,
-                          int _number, bool small ) : QWidget( _parent ), p_intf( _p_intf )
+                          int _number, bool b_small ) : QWidget( _parent ), p_intf( _p_intf )
 {
     module_config_t *p_config;
     ConfigControl *control;
@@ -709,7 +709,7 @@ SPrefsPanel::SPrefsPanel( intf_thread_t *_p_intf, QWidget *_parent,
     panel_layout->addWidget( panel_label );
     panel_layout->addWidget( title_line );
 
-    if( small )
+    if( b_small )
     {
         QScrollArea *scroller= new QScrollArea;
         scroller->setWidget( panel );
diff --git a/modules/gui/qt4/dialogs/preferences.cpp b/modules/gui/qt4/dialogs/preferences.cpp
index e7fd4c5..e3c1a5d 100644
--- a/modules/gui/qt4/dialogs/preferences.cpp
+++ b/modules/gui/qt4/dialogs/preferences.cpp
@@ -67,13 +67,13 @@ PrefsDialog::PrefsDialog( QWidget *parent, intf_thread_t *_p_intf )
     types->setAlignment( Qt::AlignHCenter );
     QHBoxLayout *types_l = new QHBoxLayout;
     types_l->setSpacing( 3 ); types_l->setMargin( 3 );
-    small = new QRadioButton( qtr( "Simple" ), types );
-    small->setToolTip( qtr( "Switch to simple preferences view" ) );
-    types_l->addWidget( small );
+    small_rb = new QRadioButton( qtr( "Simple" ), types );
+    small_rb->setToolTip( qtr( "Switch to simple preferences view" ) );
+    types_l->addWidget( small_rb );
     all = new QRadioButton( qtr("All"), types ); types_l->addWidget( all );
     all->setToolTip( qtr( "Switch to full preferences view" ) );
     types->setLayout( types_l );
-    small->setChecked( true );
+    small_rb->setChecked( true );
 
     /* Tree and panel initialisations */
     advanced_tree = NULL;
@@ -129,7 +129,7 @@ PrefsDialog::PrefsDialog( QWidget *parent, intf_thread_t *_p_intf )
     BUTTONACT( cancel, cancel() );
     BUTTONACT( reset, reset() );
 
-    BUTTONACT( small, setSmall() );
+    BUTTONACT( small_rb, setSmall() );
     BUTTONACT( all, setAdvanced() );
 
     resize( 780, sizeHint().height() );
@@ -224,7 +224,7 @@ void PrefsDialog::setSmall()
     }
 
     current_simple_panel->show();
-    small->setChecked( true );
+    small_rb->setChecked( true );
 }
 
 /* Switching from on simple panel to another */
@@ -300,7 +300,7 @@ void PrefsDialog::showModulePrefs( char *psz_module )
 /* Actual apply and save for the preferences */
 void PrefsDialog::save()
 {
-    if( small->isChecked() && simple_tree->isVisible() )
+    if( small_rb->isChecked() && simple_tree->isVisible() )
     {
         msg_Dbg( p_intf, "Saving the simple preferences" );
         for( int i = 0 ; i< SPrefsMax; i++ ){
diff --git a/modules/gui/qt4/dialogs/preferences.hpp b/modules/gui/qt4/dialogs/preferences.hpp
index 9ae0fd4..89508e3 100644
--- a/modules/gui/qt4/dialogs/preferences.hpp
+++ b/modules/gui/qt4/dialogs/preferences.hpp
@@ -69,7 +69,8 @@ private:
     SearchLineEdit *tree_filter;
 
     QGroupBox *types;
-    QRadioButton *small,*all;
+    QRadioButton *small_rb;
+    QRadioButton *all;
 
     bool b_small;
 
-- 
1.7.5.4

